test/framework/integration_framework/integration_test_framework.cpp

```diff
@@ -224,6 +224,73 @@ namespace integration_framework {
     return *this;
   }
 
+  IntegrationTestFramework &IntegrationTestFramework::sendTxSequence(
+      const shared_model::interface::TransactionSequence &tx_sequence,
+      std::function<void(std::vector<shared_model::proto::TransactionResponse>
+                             &)> validation) {
+    log_->info("send transactions");
+    const auto &transactions = tx_sequence.transactions();
+
+    boost::barrier bar(2);
+
+    iroha_instance_->instance_->getStatusBus()
+        ->statuses()
+        .filter([&transactions](auto s) {
+          // filter statuses for transactions from sequence
+          auto it = std::find_if(
+              transactions.begin(), transactions.end(), [&s](const auto tx) {
+                // check if status is either stateless valid or failed
+                bool is_stateless_status = iroha::visit_in_place(
+                    s->get(),
+                    [](const shared_model::interface::StatelessFailedTxResponse
+                           &stateless_failed_response) { return true; },
+                    [](const shared_model::interface::StatelessValidTxResponse
+                           &stateless_valid_response) { return true; },
+                    [](const auto &other_responses) { return false; });
+                return is_stateless_status
+                    and s->transactionHash() == tx->hash();
+              });
+          return it != transactions.end();
+        })
+        .take(transactions.size())
+        .subscribe([](auto &&) {}, [&bar] { bar.wait(); });
```

Comment:
Unfortunately, it seems that `getTxStatus` call will not work in a deterministic way. Consider the following scenario:
1. `ListTorii` is called, thread switch to gRPC, transactions are executed, `wait` on line 256 is called
2. thread switch to test, `wait` on line 270 is called, it continues execution because the barrier requirement of 2 is satisfied
3. thread switch to gRPC, it continues to run up to stateful validation
4. thread switch to test, `getTxStatus` is executed, but the results are stateful valid/invalid already

Therefore, it seems necessary to capture the statuses in `statuses` subscription. ` statuses.push_back(status);` can be moved inside `OnNext` lambda, and the barrier will be placed between `ListTorii` and `validation` calls.